Documentation/api-reference.md

```diff
@@ -68,72 +71,193 @@ m.aepSdk = AdobeAEPSDKInit()
+shutdown: function() as void
 ```
 
 ##### Example
 
 ```brightscript
-m.aepSdk.resetIdentities()
+m.aepSdk.shutdown();
 ```
 
 ---
 
+### updateConfiguration
+
+> **Note**
+> Some public APIs need valid configuration to process the data and make the network call to Adobe Experience Edge Network. All the hits will be queued if no valid configuration is found. It is ideal to call updateConfiguration API with valid require configuration before any other public APIs.
+
+#### Configuration Keys
+
+- Required for all APIs
+
+| Constants | Raw value | Type | Required |
+| :-- | :--: | :--: | :--: |
+| `ADB_CONSTANTS.CONFIGURATION.EDGE_CONFIG_ID` | "edge.configId" | String | **Yes**
+| `ADB_CONSTANTS.CONFIGURATION.EDGE_DOMAIN` | "edge.domain" | String | **No**
+| `ADB_CONSTANTS.CONFIGURATION.CONSENT_DEFAULT` | "consent.default" | Map | **No**
+
+- Required for Media tracking APIs
+
+| Constants | Raw value | Type | Required |
+| :-- | :--: | :--: | :--: |
+| `ADB_CONSTANTS.CONFIGURATION.MEDIA_CHANNEL` | "edgemedia.channel" | String | **Yes**
+| `ADB_CONSTANTS.CONFIGURATION.MEDIA_PLAYER_NAME` | "edgemedia.playerName" | String | **Yes**
+| `ADB_CONSTANTS.CONFIGURATION.MEDIA_APP_VERSION` | "edgemedia.appVersion" | String | **No**
+
+##### Syntax
+
+```brightscript
+updateConfiguration: function(configuration as object) as void
+```
+
+- `@param configuration as object`
+
+##### Example
+
+```brightscript
+ADB_CONSTANTS = AdobeAEPSDKConstants()
+
+configuration = {}
+configuration[ADB_CONSTANTS.CONFIGURATION.EDGE_CONFIG_ID] = "<YOUR_CONFIG_ID>"
+configuration[ADB_CONSTANTS.CONFIGURATION.EDGE_DOMAIN] = "<YOUR_DOMAIN_NAME>"
+
+' Set the default consent to pending and update the consent later based on user preference
+configuration[ADB_CONSTANTS.CONFIGURATION.CONSENT_DEFAULT] = {
+    "consents": {
+        "collect": {
+            "val": "p"
+        }
+    }
+}
+
+m.aepSdk.updateConfiguration(configuration)
+```
+
+The `EDGE_CONFIG_ID` value is presented as `Datastream ID` in the [Datastream details](https://experienceleague.adobe.com/docs/experience-platform/edge/datastreams/configure.html?lang=en#view-details) page.
+
+The `EDGE_DOMAIN` value is the first-party domain mapped to the Adobe-provisioned Edge Network domain. For more information, see this [documentation](https://developer.adobe.com/client-sdks/documentation/edge-network/#domain-configuration)
+
+## Consent APIs:
+
+### Configure default consent
+
+The default consent configuration determines how data collection consent is managed before invoking the [setConsent](#setconsent) API. This setting is particularly important to avoid unintentionally collecting data from users in regions where consent is required prior to data collection.
+
+By default, users are opted in for all purposes.
+
+#### Example: Setting default collect consent to pending before collecting actual user consent
+```brightscript
+ADB_CONSTANTS = AdobeAEPSDKConstants()
+
+configuration = {}
+
```

Comment:
This example sets the default consent to pending. You can later update the consent based on user preferences using the setConsent API.